okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpWritableBufferAllocator.java

```diff
@@ -38,8 +38,24 @@
  * The default allocator for {@link OkHttpWritableBuffer}s used by the OkHttp transport.
  */
 class OkHttpWritableBufferAllocator implements WritableBufferAllocator {
+
+  private final int maxDataLength;
+
+  /**
+   * Construct a new instance
+   * @param maxDataLength the maximum allowed length of a DATA frame.
```

Comment:
Instead of limiting here, could we do min(maxBytes, maxDataLength) in [OutboundFlowController.OutboundFlowState.writeBytes()](https://github.com/grpc/grpc-java/blob/af188767134e387a798803230c730d9f99ed214f/okhttp/src/main/java/io/grpc/transport/okhttp/OutboundFlowController.java#L316)